### ISSUE TITLE
Rule blocked: CORERULES-15 CORERULES-16

### ISSUE THREAD

**Issue by gcampion-cdisc** [cdisc-org/cdisc-rules-engine#221]
This bug was created in an ADO PBI but was not carried to Github until now

**Sample Request**
[cg0037.json.txt](https://github.com/cdisc-org/cdisc-rules-engine/files/10115132/cg0037.json.txt)

**Links to related JIRA Tickets**
- https://dev.azure.com/cdisc-org/cdisc-core/_workitems/edit/4708
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-15
- https://jira.cdisc.org/projects/CORERULES/issues/CORERULES-16

**Rule Information**
- **Standard**: SDTMIG
- **Rule ID**: CG0037, CG0039
- **Rule Description**: --SOCCD != --BDSYCD

**Describe the bug**
Rule Sample:
```yaml
- name: --SOCCD
  operator: not_equal_to
  value: --BDSYCD
```
The response is unexpected because we are comparing null not_equal_to null, which should return false instead of true.

**Error returned from Rule Engine**
```json
{
  "MH": [
    {
      "domain": "MH",
      "errors": [
        {
          "row": 4,
          "seq": 1,
          "uSubjId": "CDISC006",
          "value": {
            "MHBDSYCD": null,
            "MHSOCCD": null,
            "MHTERM": "ALZHEIMER'S DISEASE"
          }
        }
      ],
      "executionStatus": "success",
      "message": "Primary analysis was used but MHSOCCD and MHBDSYCD are not equal",
      "variables": [
        "MHBDSYCD",
        "MHSOCCD",
        "MHTERM"
      ]
    }
  ]
}
```

**Expected behavior**
- When doing "equal_to" and "not_equal_to" and other comparison type operators, empty "" string values should be handled the same way null numeric values are handled.
- The attached request should return no results in the "errors" array.
- The comparisons should return false for both "equal_to" and "not_equal_to" whenever both comparators are empty "" or null.
- The comparisons should follow this truth table:
https://cdisc-org.github.io/conformance-rules-editor/#/boolean_logic?id=equal_to-not_equal_to

**Notes**
I believe [#3968](https://dev.azure.com/cdisc-org/2c53d232-12ce-401e-9084-84689e2d7372/_workitems/edit/3968)  partially resolved the issue, but maybe the issue was only resolved for the case of empty "" string values. In this rule, we have null numeric values instead.

**Comment by nhaydel** [comment]
@gcampion-cdisc how does this block these rules and not other rules that use equal_to and not_equal_to? This seems more like a general bug with the engine than a rule blocking bug

**Comment by gcampion-cdisc** [comment]
> @gcampion-cdisc how does this block these rules and not other rules that use equal_to and not_equal_to? This seems more like a general bug with the engine than a rule blocking bug

@nhaydel If there are other rules that use equal_to/not_equal_to _on variables that are numeric_, then those should also be blocked by this. Those rules should not pass QC if they can't be tested completely.
Either these operators haven't been used for numeric variables in other rules or the authors didn't include test cases for null-null for those rules.

**Comment by nhaydel** [comment]
@gcampion-cdisc that is my point, I am almost positive that there are published rules that use these operators on non numeric data, should they be unpublished or should this instead be classified as a rule engine bug rather than blocking rules from being published since it is an issue with the engine itself, not the rule

**Comment by gcampion-cdisc** [comment]
@nhaydel 
I see what you are saying.
On the one hand, those rules should be unpublished because they would flag a bunch of false negatives with real data.
On the other hand, the rule content likely won't be changing to fix this bug - just the engine.
I guess it's a decision for the rule publishers.

For the rule authors, classifying this as a rule-blocking bug with jira id's allows them to track the bug status with the rules (if the branches or prs are created containing the same ids). What is the impact on your side of classifying this as a rule-blocking bug instead of a general bug?
